config/routes.rb

```diff
@@ -0,0 +1,29 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  root "homepages#index"
+  get "/search", to: "homepages#search", as: "search"
+  
+  get "/cart", to: "orders#cart", as: "cart"
+  get "/checkout/:id", to: "orders#edit", as: "checkout"
+  resources :orders, except: [:index, :new, :create, :put, :delete]
+  
+  resources :merchants, only: [:index, :show]
+  post "/logout", to: "merchants#logout", as: "logout"
+  get "/dashboard", to: "merchants#dashboard", as: "dashboard"
+  
+  resources :orderitems, except: [:index, :put, :new, :show]
+  patch "/orderitems/:id/cancel/", to: "orderitems#cancel", as: "cancel"
+  
+  resources :products, except: [:put]
```

Comment:
You don't do except `put` you would do except update or something along those lines.